test/e2e/tests/notebook/notebook-create.test.ts

```diff
@@ -37,6 +44,58 @@ test.describe('Notebooks', {
 			await app.workbench.notebooks.stopEditingCell();
 			await app.workbench.notebooks.assertMarkdownText('h2', randomText);
 		});
+
+		test('Python - Save untitled notebook and preserve session', async function ({ app }) {
+			// Ensure auxiliary sidebar is open to see variables pane
+			await app.workbench.layouts.enterLayout('notebook');
+			await app.workbench.quickaccess.runCommand('workbench.action.toggleAuxiliaryBar');
+
+			// First, create and execute a cell to verify initial session
+			await app.workbench.notebooks.addCodeToFirstCell('foo = "bar"');
+			await app.workbench.notebooks.executeCodeInCell();
+
+			// Verify the variable is in the variables pane
+			await app.workbench.variables.expectVariableToBe('foo', "'bar'");
+
+			// Save the notebook using the command
+			await app.workbench.quickaccess.runCommand('workbench.action.files.saveAs', { keepOpen: true });
+			await app.workbench.quickInput.waitForQuickInputOpened();
+			// Wait a bit for the save to complete
+			// await new Promise(resolve => setTimeout(resolve, 1000));
+			// Generate a random filename
+			const newFileName = `saved-session-test-${Math.random().toString(36).substring(7)}.ipynb`;
+
+			await app.workbench.quickInput.type(path.join(app.workspacePathOrFolder, newFileName));
+			await app.workbench.quickInput.clickOkButton();
+
+			// Wait a bit for the save to complete
+
+			// Wait a bit for the save to complete
+			await new Promise(resolve => setTimeout(resolve, 1000));
```

Comment:
Is there nothing in the UI that we can wait for that would be more robust + faster?

Reply:
There is! Fixed.

Reply:
Actually I realized the waiting will automatically happen for us because of the next line.